pkg/utils/split.go

```diff
@@ -11,3 +14,95 @@ var typesForPkgSplit = []ftypes.TargetType{ftypes.NodePkg, ftypes.Yarn, ftypes.N
 func IsPkgSplitRequired(targetType ftypes.TargetType) bool {
 	return lo.Contains(typesForPkgSplit, targetType)
 }
+
+// Func splits the given licenses (if any separators like AND/OR are present in the license),
+// validates those individual licenses if they are SPDX defined or not.
+// If any SPDX license is found, it filters those licenses and returns them.
+// else we just return the given input license string
+func FilterNGetLicenses(licenses []string) []string {
+	var foundSPDXLicense bool
+	var result []string
+	var tempLicenses []string
+
+	// Split and validate licenses
+	for _, license := range licenses {
+		// skip empty licenses
+		if len(license) == 0 {
+			continue
+		}
+		tempLicenses = append(tempLicenses, license)
```

Comment:
why is this required?

Reply:
i'm updating the license to this temp licenses array. This is to filter out any empty license strings present 
 
`licenses = tempLicenses
	if !foundSPDXLicense {
		return lo.Uniq(licenses)
	}`

Reply:
Hmm. Can't this be handle when we create the licenses array?

Reply:
Can be but it's more effort. Since licenses are populated for different languages in different packages, i need to modify those as well. (i.e not just node, dotnet related things). for now, i'm handling them in post processing. @namandf